_source/_posts/2021-10-16-full-stack-java.md

```diff
@@ -0,0 +1,693 @@
+---
+layout: blog_post
+title: "Full Stack Java with React, Spring Boot, and JHipster"
+author: matt-raible
+by: advocate
+communities: [java,javascript]
+description: "This tutorial shows you how to be a full stack Java developer with React and Spring Boot."
+tags: [java, full-stack, react, spring-boot, jhipster]
+tweets:
+- ""
+- ""
+- ""
+image:
+type: conversion
+---
+
+If you search for "Full Stack Java" on the internet, you'll likely find a lot of recruiting, courses, and jobs. Being a full stack developer can be exciting because you're learning and using a bunch of technologies. It also pays pretty well. Today, I'm going to show you how you can be a full stack Java developer with Spring Boot, React, and JHipster.
+
+If you haven't heard of JHipster, boy do I have a treat for you! JHipster started as a Yeoman application generator back in 2013 and has grown to become a development platform. It allows you to quickly generate, develop, and deploy modern web apps and microservice architectures. Today, I'll show you how to build a Flickr clone with JHipster and lock it down with OAuth and OpenID Connect (OIDC).
+
+**Prerequisites:**
+
+- [Node.js 14+](https://nodejs.org/)
+- [Java 11+](https://sdkman.io)
+
+**Table of Contents**{: .hide }
+* Table of Contents
+{:toc}
+
+## Full Stack Development with React and Spring Boot
+
+One of the easiest ways to get started with React is by using [Create React App](https://create-react-app.dev/) (CRA). You install it locally, then run `create-react-app <project>` to generate a React application with minimal dependencies. It uses webpack under-the-covers to build the project, launch a web server, and run its tests.
+
+Spring Boot has a similar tool, called [Spring Initializr](https://start.spring.io/). Spring Initializer is a bit different than CRA because it's a website (and API) that you can create applications with.
```

Comment:
```suggestion
Spring Boot has a similar tool called [Spring Initializr](https://start.spring.io/). Spring Initializer is a bit different than CRA because it's a website (and API) that you can create applications with.
```